### PR TITLE
add note about HelpLink and clean up another admonition

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-12668

### Staged Changes (Requires MongoDB Corp SSO)

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/12668/sdk/dotnet/examples/sync-changes-between-devices/#handle-sync-errors

### CODE REVIEW

File: source/sdk/dotnet/examples/sync-changes-between-devices.txt [hunk 1]
```diff
@@ -168,6 +168,13 @@ strongly typed handling logic.
 .. literalinclude:: /examples/generated/dotnet/ErrorHandler.codeblock.handle-errors.cs
    :language: csharp
 
+.. note:: Additional Exception Information
+
+   For security reasons, {+service-short+} may send only a minimal amount of 
+   information about an exception, but the server-side logs will contain more
```
Comment (nlarew) [inline source/sdk/dotnet/examples/sync-changes-between-devices.txt:174]: Consider a few tweaks, up to you to include or not

```rst
For security reasons, {+service-short+} sends a minimal description of
exceptions to the SDK. To see more detailed exception descriptions,
refer to your app's :ref:`server-side error logs <logs>`.
```

File: source/sdk/dotnet/examples/sync-changes-between-devices.txt [hunk 2]
```diff
@@ -149,11 +149,11 @@ Once you no longer wish to receive notifications, unregister the token by using
 
 .. warning::
```
Comment (nlarew) [inline source/sdk/dotnet/examples/sync-changes-between-devices.txt:150]:
```suggestion
.. note:: 
```

This doesn't really feel like a warning to me, but rather a potentially useful thing to know.

Same with the warning above it that's not in this diff.

File: source/sdk/dotnet/examples/sync-changes-between-devices.txt [hunk 3]
```diff
@@ -168,6 +168,13 @@ strongly typed handling logic.
 .. literalinclude:: /examples/generated/dotnet/ErrorHandler.codeblock.handle-errors.cs
    :language: csharp
 
+.. note:: Additional Exception Information
+
+   For security reasons, {+service-short+} may send only a minimal amount of 
+   information about an exception, but the server-side logs will contain more 
+   details. In these cases, the 
+   `HelpLink <https://docs.microsoft.com/en-us/dotnet/api/system.exception.helplink?view=net-6.0#system-exception-helplink>`_
```
Comment (nlarew) [inline source/sdk/dotnet/examples/sync-changes-between-devices.txt:176]: **Non-blocking**
Should we add a role to `rstspec.toml` for this? e.g. [`:azure:`](https://github.com/mongodb/snooty-parser/blob/master/snooty/rstspec.toml#L1124)

```toml
[role.ms-dotnet]
help = "Link to the Microsoft .NET documentation"
type = {link = "https://docs.microsoft.com/en-us/dotnet%s"}
```


```rst
:ms-dotnet:`HelpLink <api/system.exception.helplink?view=net-6.0#system-exception-helplink>`
```

(Maybe without the query param and hash if possible)

**Review by nlarew — APPROVED** [review]
LGTM w/ a few comments for your consideration

File: source/sdk/dotnet/examples/sync-changes-between-devices.txt (hunk 3 above)
Comment (MongoCaleb) [reply source/sdk/dotnet/examples/sync-changes-between-devices.txt:176]: I suppose we could, though doesn't the rstspec.toml live in a separate repo? Also, we have several other instances, some of which point to /xamarin or /visualstudio.

**Pull request merged**